### DISCUSSION TITLE
How to set badge on checkbox?

### DISCUSSION THREAD

**Discussion by tz301** [zauberzeug/nicegui#3218] [Q&A] [answered]
### Question

From https://nicegui.io/documentation/badge, badge can be set on button.

```py
with ui.button('Warning!'):
    badge = ui.badge('!', color='red').props('floating')
```

![image](https://github.com/zauberzeug/nicegui/assets/35760938/8b0a94d7-d566-4178-8ec7-c1c8a0d3bd34)

However, when I set badge on checkbox, it seems not working. It will be set on outside context, not on target checkbox.

```py
with ui.card():
    with ui.checkbox('Warning!'):
        badge = ui.badge('!', color='red').props('floating')

    ui.checkbox('Select')
```

![image](https://github.com/zauberzeug/nicegui/assets/35760938/ef77fb86-ad1f-4f84-8303-a8fccfafffe1)

**Comment by falkoschindler** [comment]
That's unexpected!
Try adding `.classes('relative')` to the checkbox element. This causes the badge to be placed relative to the checkbox rather than the card. (The way HTML elements are placed based on their CSS "position" attribute is pretty confusing, but you'll find plenty resources in case you want to know more.)

**Reply by tz301** [reply]
Hi @falkoschindler ,

Yeah, that works.

![image](https://github.com/zauberzeug/nicegui/assets/35760938/2f5696ad-f4b3-4eec-ada5-3bf210325e45)

However, when I use w-full to place the container, the badge seems weird. Any attribute to set it close to the checkbox?

```py
with ui.card().classes('w-full'):
    with ui.grid(columns=2):
        with ui.checkbox('Warning!').classes('relative'):
            badge = ui.badge('!', color='red').props('floating')

        ui.checkbox('Select')
```

![image](https://github.com/zauberzeug/nicegui/assets/35760938/e48cbbf0-7b8b-4c10-acb3-e794fdd71607)

**Reply by falkoschindler** [reply, marked as answer]
In your example the checkbox is pretty wide, which you can verify by adding a border or background color to `ui.checkbox`. You can bring it back to its original size and stay left-aligned using the "justify-items-start" class:
```py
with ui.card().classes('w-full'):
    with ui.grid(columns=2).classes('w-full justify-items-start'):
        with ui.checkbox('Warning!').classes('relative border'):
            badge = ui.badge('!', color='red').props('floating')
        ui.checkbox('Select').classes('border')
```
<img width="692" alt="Screenshot 2024-06-14 at 07 31 03" src="https://github.com/zauberzeug/nicegui/assets/5767091/4bcd1683-8ccd-4a99-8d2e-8f1c76505dcd">

**Reply by tz301** [reply]
Thanks!
